### PR TITLE
Optionally use an image as map instead of rendering terrain

### PR DESCRIPTION
**Pull request by t0stiman**
This is useful for custom maps that don't use the terrain scene or to make prettier maps.

### CODE REVIEW

File: MapifyEditor/Export/Validators/Project/MapInfoValidator.cs [hunk 1]
```diff
@@ -42,6 +42,18 @@ protected override IEnumerator<Result> Validate(Scenes scenes)
                 yield return Result.Error($"The spawn position's Y value must be above the terrain ({worldHeight})", mapInfo);
             if (spawnPos.y < mapInfo.waterLevel)
                 yield return Result.Error($"The spawn position must be above the water level ({mapInfo.waterLevel}", mapInfo);
+
+            if (mapInfo.useFixedMapImage)
+            {
+                if (mapInfo.fixedMapImage == null)
+                {
+                    yield return Result.Error("Mapinfo 'fixedMapImage' must be set when 'useFixedMapImage' is true", mapInfo);
```
Comment (Insprill) [inline MapifyEditor/Export/Validators/Project/MapInfoValidator.cs:50]: Please use `nameof` for class/variable names

File: MapifyEditor/Export/Validators/Project/MapInfoValidator.cs [hunk 2]
```diff
@@ -42,6 +42,18 @@ protected override IEnumerator<Result> Validate(Scenes scenes)
...
+                }
+                else if(mapInfo.fixedMapImage.width != mapInfo.fixedMapImage.height)
+                {
+                    yield return Result.Warning($"Mapinfo 'fixedMapImage' should be square or it will be stretched. Current dimensions: {mapInfo.fixedMapImage.width}x{mapInfo.fixedMapImage.height}", mapInfo);
```
Comment (Insprill) [inline MapifyEditor/Export/Validators/Project/MapInfoValidator.cs:54]: Same here

**Review by Insprill — CHANGES_REQUESTED** [review]

File: MapifyEditor/Export/Validators/Project/MapInfoValidator.cs (hunk 1 above)
Comment (t0stiman) [reply MapifyEditor/Export/Validators/Project/MapInfoValidator.cs]: fixed

File: MapifyEditor/Export/Validators/Project/MapInfoValidator.cs (hunk 2 above)
Comment (t0stiman) [reply MapifyEditor/Export/Validators/Project/MapInfoValidator.cs]: fixed

**Comment by t0stiman** [comment]
This one is ready for review

**Review by Insprill — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**